src/api/firebase.js

```diff
@@ -287,3 +291,25 @@ export async function getListOwnerDetails(userId, listName) {
 		throw error;
 	}
 }
+
+export async function deleteSingleList(userEmail, listPath) {
+	const userRef = doc(db, 'users', userEmail);
+	const listRef = doc(db, listPath);
+
+	const listDoc = await getDoc(listRef);
+	const userDoc = await getDoc(userRef);
+
+	if (listDoc.data().ownerID !== userDoc.data().uid) {
+		throw new Error('You are not the owner of this list.');
+	}
+
+	try {
+		await deleteDoc(listRef);
+		updateDoc(userRef, {
+			sharedLists: arrayRemove(listRef),
+		});
+		return { success: true, message: 'List successfully deleted' };
+	} catch (error) {
+		throw new Error(`Could not delete currunt list. An error: ${error}`);
+	}
```

Comment:
`nitpick` - there's a small typo in the word "current".

Reply:
Thanks! Fixed!